xchange-stream-coinjar/src/main/java/info/bitrich/xchangestream/coinjar/CoinjarStreamingService.java

```diff
@@ -0,0 +1,51 @@
+package info.bitrich.xchangestream.coinjar;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.coinjar.dto.CoinjarHeartbeat;
+import info.bitrich.xchangestream.coinjar.dto.CoinjarWebSocketSubscribeMessage;
+import info.bitrich.xchangestream.coinjar.dto.CoinjarWebSocketUnsubscribeMessage;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.reactivex.Observable;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+class CoinjarStreamingService extends JsonNettyStreamingService {
+
+  private final AtomicInteger refCount = new AtomicInteger();
+
+  private String apiKey;
+
+  public CoinjarStreamingService(String apiUrl, String apiKey) {
+    super(apiUrl);
+    this.apiKey = apiKey;
+    this.subscribeConnectionSuccess()
```

Comment:
I don't this is is the right way to set this up because if you have disconnect and reconnect this will publish a message for every reconnect you have had during the course of your application. So if you disconnect 2 times and reconnect a 3rd this will publish your message 3 times a second when you reconnect.

You can test this by breaking internet connection deliberately and seeing how many time the message gets published. Something like: 

`    Observable.interval(1, TimeUnit.SECONDS)
        .subscribe(
            t -> {
              if (isSocketOpen()) webSocketChannel.writeAndFlush(new PingWebSocketFrame());
            });`

Should accomplish what you want

Reply:
Done, good catch.